src/Workspaces/CoreTest/SymbolKeyTests.cs

```diff
@@ -113,6 +114,28 @@ public class B { };
             TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70782")]
+        public async Task TestNintNuint()
+        {
+            var source = @"
+
+public class C
+{
+    void M(nint x);
+    void N(nuint x);
+}
+";
+            var netstandardReferences = await ReferenceAssemblies.NetStandard.NetStandard20.ResolveAsync(LanguageNames.CSharp, cancellationToken: default);
+            var netcoreReferences = await ReferenceAssemblies.Net.Net70.ResolveAsync(LanguageNames.CSharp, cancellationToken: default);
+
+            var compilation1 = GetCompilation(source, LanguageNames.CSharp, references: [.. netstandardReferences]);
+            var compilation2 = GetCompilation(source, LanguageNames.CSharp, references: [.. netcoreReferences]);
+            TestRoundTrip(GetDeclaredSymbols(compilation1), compilation1, useSymbolEquivalence: false);
+            TestRoundTrip(GetDeclaredSymbols(compilation1), compilation2, useSymbolEquivalence: true);
+            TestRoundTrip(GetDeclaredSymbols(compilation2), compilation1, useSymbolEquivalence: true);
+            TestRoundTrip(GetDeclaredSymbols(compilation2), compilation2, useSymbolEquivalence: false);
```

Comment:
across the same compilation, they should be .Equals.  Across different compilations, then SymbolEquivalence should match.